third-party/qthread/Makefile

```diff
@@ -144,33 +127,17 @@ endif
 ifneq (, $(CHPL_QTHREAD_SCHEDULER))
 SCHEDULER = $(CHPL_QTHREAD_SCHEDULER)
 endif
-CHPL_QTHREAD_CFG_OPTIONS += --with-scheduler=$(SCHEDULER)
+CHPL_QTHREAD_CFG_OPTIONS += -DQTHREADS_SCHEDULER=$(SCHEDULER)
 
 # only do this on M1+ Macs
 ifeq ($(CHPL_MAKE_TARGET_ARCH),arm64)
 ifeq ($(CHPL_MAKE_TARGET_PLATFORM),darwin)
-  CHPL_QTHREAD_CFG_OPTIONS += --with-cacheline-width=128
+  CHPL_QTHREAD_CFG_OPTIONS += -DQTHREADS_CACHELINE_SIZE_ESTIMATE=128
 endif
 endif
 
 # reduce performance penalty in cases where numChapelTasks < numQthreadWorkers
-CHPL_QTHREAD_CFG_OPTIONS += --enable-condwait-queue
-
-# pgi doesn't support the proper atomic intrinsics that are required for
-# spinlocks to be fast, so disable them for pgi, and PrgEnv-pgi
-ifneq (, $(findstring pgi,$(CHPL_MAKE_TARGET_COMPILER)))
-CHPL_QTHREAD_CFG_OPTIONS += --disable-internal-spinlock
-endif
```

Comment:
does this option no longer exist? does the pgi issue still stand?

Reply:
We no longer support PGI since it hasn't been under development for a while.

Reply:
As fair as I know, Chapel still supports PGI (although I am pretty certain we don't test it nightly). Does `--disable-internal-spinlock` no longer exist in any form?

To be clear, in my opinion this is not a big deal since PGI is not actively developed and probably isn't that important (and based on the comments in the code, this is a perf issue not a correctness issue?) Just want to make sure we aren't potentially dropping something on the floor thats still usable and thats still useful to users

Reply:
It doesn't exist anymore, but it's not terribly hard to re-add it. On the other hand, I'm not sure the comment about PGI not supporting intrinsics applies anymore either. They've likely added support for them since the comment was written. Also, the qthreads exposure to anything other than c11 atomics has decreased pretty dramatically, though there are still a few intrinsics left that could still cause trouble.

Reply:
That is good enough for me, no need to re-add it. I'm fine with deleting this special case